doc/doxygen/headers/automatic_and_symbolic_differentiation.h

```diff
@@ -0,0 +1,97 @@
+//
+// Copyright (C) 2017 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+/**
+ * @defgroup auto_symb_diff Automatic and symbolic differentiation
+ *
+ * @brief A module dedicated to the implementation of functions and classes that relate
+ * to automatic and symbolic differentiation.
+ *
+ * @todo Hyper-summarize the following list of topics:
+ * - Automatic differentiation
+ * - Symbolic differentiation
+ *
+ * @section auto_diff_1 Automatic differentiation
+ *
+ * @todo Write a short introduction into AD. As a temporary entry, the following links
+ * may be enlightening:
+ * - <a href="https://en.wikipedia.org/wiki/Automatic_differentiation">Wikipedia article</a>
+ * - <a href="https://projects.coin-or.org/ADOL-C/browser/stable/2.6/ADOL-C/doc/adolc-manual.pdf?format=raw#page=1>Adol-C manual</a>
+ *
+ * @todo Hyper-summarize the following list of topics:
+ * - Forward and reverse mode AD
+ * - Taped and tapeless AD
+ *
+ * @subsection auto_diff_1_1 Supported automatic differentiation libraries
+ *
+ * We currently have validated implementations for the following number types
+ * and combinations:
+ *
+ *  - Taped Adol-C (n-differentiable, in theory, but internal drivers for up to second-order
+ *    derivatives have been implemented)
+ *  - Tapeless Adol-C (once differentiable)
+ *  - Tapeless forward-mode Sacado with dynamic memory allocation (once differentiable)
+ *  - Tapeless nested forward-mode Sacado (twice differentiable)
+ *  - Tapeless reverse-mode Sacado (once differentiable)
+ *  - Tapeless nested reverse and forward-mode Sacado (twice differentiable)
+ *
+ * @subsection auto_diff_1_2 How automatic differentiation is integrated into deal.II
+ *
+ * Since the interface to each automatic differentiation library is so vastly different,
+ * a uniform internal interface to each number has been established. This allows the
+ * driver classes (that provide the core functionality, and are introduced in the next
+ * section) a consistent mechanism to interact with different auto-differentiation
+ * libraries. Specifically, they need to be able to correctly initialize and finalize data
+ * that is to be interpreted as the dependent and independent variables of a formula.
+ *
+ * A summary of the files that implement the interface to the supported auto-differentiable
+ * numbers is as follows:
+ *
+ * - ad_number_types.h : Introduces an enumeration (called a type code) for the
```

Comment:
no space before the colon. there are numerous other places here where this is the case